### DISCUSSION TITLE
Is there a way to mark responses as ones that should be sent base64 encoded to be returned via api gateway?

### DISCUSSION THREAD

**Discussion by tqhoughton** [awslabs/aws-lambda-web-adapter#452] [Q&A] [answered]
Using a raw lambda proxy integration you can do something like:

```ts
return {
  statusCode: 200,
  body: myBase64EncodedImage,
  isBase64Encoded: true
}
```

and you are able to send binary files back via API Gateway (up to 10mb).

Does lambda web adapter expose functionality to do the same thing somehow or is this not supported? I didn't see base64 encoding called out anywhere so I don't know if I missed it or not.

My current lambda service is able to return html and css just fine but when I try to return an image that is bundled with my container the content of the body seems to get stripped out.

**Comment by bnusunny** [comment, marked as answer]
Lambda Web Adapter does support binary contents. LWA uses lambda-http crate to handle that. Lambda-http will detect binary contents based on content-type and content-encoding headers, and auto base64 encode the data. You don't need to encode the binary data yourself. 

[Here](https://github.com/awslabs/aws-lambda-rust-runtime/blob/de822f9d870c21c06b504d218293099f691ced9f/lambda-http/src/response.rs#L25-L36) are the content-type prefixes and suffixes that are treated as text. Others are treated as binary.

**Comment by bnusunny** [comment]
You can take a look at the [Nginx example](https://github.com/awslabs/aws-lambda-web-adapter/tree/main/examples/nginx).

**Comment by tqhoughton** [comment]
Thank you for the links to the code, was able to gleam from there that anything with a proper `Content-Encoding: gzip` header key/value gets returned properly!

In my express app it looks something like this, if anyone else runs into this issue:

```ts
import express from 'express'
import compression from 'compression'

...
const shouldCompressResponse = (req: express.Request) => {
  // compress images so apigateway can return them w/ binary media types
  logger.info('extension: ', path.extname(req.url))
  return ['.jpg', '.png', '.ico'].includes(path.extname(req.url))
}

app.use('/ui/assets', compression({ filter: shouldCompressResponse }))
app.use('/ui/assets', express.static('public/assets/'))
```
